src/Common/atomicRename.cpp

```diff
@@ -97,9 +101,9 @@ static bool renameat2(const std::string & old_path, const std::string & new_path
     ErrnoException::throwFromPath(ErrorCodes::SYSTEM_ERROR, new_path, "Cannot rename {} to {}", old_path, new_path);
 }
 
-bool supportsAtomicRename()
+bool supportsAtomicRename(std::string * out_message)
 {
-    static bool supports = supportsAtomicRenameImpl();
+    static bool supports = supportsAtomicRenameImpl(out_message);
```

Comment:
I think `out_message` won't be populated after the first invocation if we don't support rename, because the local variable is static local variable, thus it will only initialized once, so `supportsAtomicRenameImpl` will only be called once.

Reply:
Oops, didn't notice the `static`, good catch.